### ISSUE TITLE
network online finished before all interfaces are initialized

### ISSUE THREAD

**Issue by MorphBonehunter** [systemd/systemd#28330]
### systemd version the issue has been seen with

253

### Used distribution

Arch Linux

### Linux kernel version used

6.1.38-1-lts

### CPU architectures issue was seen on

x86_64

### Component

systemd-networkd, systemd-networkd-wait-online

### Expected behaviour you didn't see

The network configuration is delayed until all interfaces are initialized.

### Unexpected behaviour you saw

The network startup and the network online target is finished bevor all interfaces are are initialized.
This leads to failed services which will bind to configured IPs as the `systemd-networkd-wait-online.service` returns to early.
This wasn't an problem until this change in v253:
* "systemd-networkd-wait-online exits successfully when all interfaces are ready or unmanaged"

### Steps to reproduce the problem

I can reproduce this every time i reboot my system (the problem occurs on 3 identical systems).
I do play with the `RequiredFamilyForOnline=ipv4` setting in the interfaces `[Link]` section, but as the interface isn't present after the system returns the "Network is Online" state, that doesn't make any difference.
I submit the relevant log lines from the boot process which shows the problem.

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
Jul 10 09:42:10 uv-tools1 systemd[1]: Starting Network Configuration...
Jul 10 09:42:10 uv-tools1 systemd-networkd[280]: lo: Link UP
Jul 10 09:42:10 uv-tools1 systemd-networkd[280]: lo: Gained carrier
Jul 10 09:42:10 uv-tools1 systemd-networkd[280]: Enumeration completed
Jul 10 09:42:10 uv-tools1 systemd[1]: Started Network Configuration.
Jul 10 09:42:10 uv-tools1 systemd[1]: Reached target Network.
Jul 10 09:42:10 uv-tools1 systemd[1]: Starting Wait for Network to be Configured...
Jul 10 09:42:10 uv-tools1 systemd-networkd-wait-online[312]: Found link lo(1)
Jul 10 09:42:10 uv-tools1 systemd-networkd-wait-online[312]: lo: link is ignored
Jul 10 09:42:10 uv-tools1 systemd[1]: Finished Wait for Network to be Configured.
Jul 10 09:42:10 uv-tools1 systemd[1]: Reached target Network is Online.
Jul 10 09:42:10 uv-tools1 kernel: r8169 0000:02:00.0 eth0: RTL8168g/8111g, 4c:52:62:ae:64:62, XID 4c0, IRQ 130
Jul 10 09:42:10 uv-tools1 kernel: r8169 0000:02:00.0 eth0: jumbo features [frames: 9194 bytes, tx checksumming: ko]
Jul 10 09:42:11 uv-tools1 kernel: r8169 0000:02:00.0 lan0: renamed from eth0
Jul 10 09:42:11 uv-tools1 systemd-networkd[280]: eth0: Interface name change detected, renamed to lan0.
Jul 10 09:42:11 uv-tools1 systemd-networkd[280]: lan0: Configuring with /etc/systemd/network/00-lan0.network.
Jul 10 09:42:11 uv-tools1 systemd-networkd[280]: lan0: Link UP
Jul 10 09:42:14 uv-tools1 systemd-networkd[280]: lan0: Gained carrier
Jul 10 09:42:15 uv-tools1 systemd-networkd[280]: lan0: Gained IPv6LL
```

**Comment by MorphBonehunter** [comment]
In my case this could be circumvent if you deaktivate the `systemd-networkd-wait-online` and enable the `systemd-networkd-wait-online@lan0`, but on systems with many interfaces this could be cumbersome.

**Comment by yuwata** [comment]
Already fixed by 2f96a29c2c55bdd67cdd8e0b0cfd6971968e4bca (#27825), and the fix has been already backported to stable repository. Please use the newest point release.
